main.tf

```diff
@@ -11,16 +11,29 @@ data "google_compute_network" "gke_vpc" {
 }
 
 resource "google_container_cluster" "a_gke_cluster" {
-  project = "${var.project}"
-  name    = "${lookup(var.gke_cluster, "name", "")}"
-  region  = "${var.region}"
-
-  private_cluster        = "${lookup(var.gke_cluster, "private_cluster", "")}"
-  network                = "${data.google_compute_network.gke_vpc.self_link}"
-  subnetwork             = "${google_compute_subnetwork.a_gke_subnet.self_link}"
-  master_ipv4_cidr_block = "${lookup(var.gke_cluster, "master_ipv4_cidr_block", "")}"
+  project    = "${var.project}"
+  name       = "${lookup(var.gke_cluster, "name", "")}"
+  region     = "${var.region}"
+  network    = "${data.google_compute_network.gke_vpc.self_link}"
+  subnetwork = "${google_compute_subnetwork.a_gke_subnet.self_link}"
+
+  private_cluster_config {
+    # Enables a private cluster, creating a private endpoint and nodes with private-only IP's.
+    enable_private_nodes = "${lookup(var.gke_cluster, "enable_private_nodes", true)}"
+
+    /* Despite what this looks like, setting enable_private_endpoint to false does not disable the
+         * private endpoint. When false the public endpoint is enabled.  If you want to disable the
+         * public endpoint for a cluster, then set this value to true.
+        */
```

Comment:
Can we rename this to `enable_public_endpoint` and reverse the value so our consumers are clearer on what they're doing? Something like `"${lookup(var.gke_cluster, "enable_public_endpoint", true) ? false : true`.

Reply:
We could ya.  I've tried to keep a general practice of keeping the argument names consistent through resource -> module , so that the module instance generated by a consumer has a clear path to each actual resource argument.  Makes hunting thing down docu wise a bit easier and to not loose where you're at traversing a module instance back through to the resource.

All of that said this one is really bad...I'll noodle changing this.